### PR TITLE
Cleanup appointments controller/service post-processing logic

### PR DESCRIPTION
**Pull request by JunTaoLuo**
## Summary

- This PR cleans up the post processing logic between different `appointments` API endpoints, which was identified as we were looking at IEN/Station ID logic. The goal is to reuse common logic as much as possible and consolidate the locations where post processing occurs.
- The merging clinic and facility details will be moved from the controller to the service.
- Appointments (VAOS) team.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/91164

## Testing done

- [x] The changes are covered by existing unit/regression tests

## Screenshots
N/A

## What areas of the site does it impact?
Appointments (VAOS)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:72]: Previously, GET appointments with specified ID always populates facilities and clinic details via the controller logic so I've kept it the same here. This is an open question whether we want to let params set this behavior like is done with the GET appointments with no specified ID.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:245]: FYI @cferris32 I expect this to conflict with your AVS PR so I'm going to hold this back until you're done.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:98]: I find it strange that POST and PUT calls do not call `prepare_appointment` and instead only perform a subset of the post processing. I have kept these separate for now but I'd like to understand whether there's a reason for this difference or whether this was an oversight. I looked through `prepare_appointment` and didn't see any obvious danger so I'm leaning towards calling `prepare_appointment` here as well. But before I do that I'd like to double check my understanding @simiadebowale @cferris32.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [inline modules/vaos/app/services/vaos/v2/appointments_service.rb:303]: Updating this to set the error message like how the function behaved in the controller.

File: modules/vaos/spec/request/v2/appointments_request_spec.rb
Comment (JunTaoLuo) [inline modules/vaos/spec/request/v2/appointments_request_spec.rb:728]: Needed to update the test to use more realistic traffic and mock facility due to logging and post processing expectations.

File: modules/vaos/spec/request/v2/appointments_request_spec.rb
Comment (JunTaoLuo) [inline modules/vaos/spec/request/v2/appointments_request_spec.rb:793]: Needed to update the test to use more realistic traffic and mock facility due to logging and post processing expectations.

File: modules/vaos/spec/services/v2/appointment_service_spec.rb
Comment (JunTaoLuo) [inline modules/vaos/spec/services/v2/appointment_service_spec.rb:95]: Since moving the merging of facility and clinic details to the service, several tests needed to use an additional cassette. The clinic ID from the appointment requests was also modified to match the one the clinic request.

**Review by JunTaoLuo — COMMENTED** [review]
A few notes.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [reply modules/vaos/app/services/vaos/v2/appointments_service.rb:72]: We discussed this during our AVS conversations. This is the expected behavior.

File: modules/vaos/app/controllers/vaos/v2/appointments_controller.rb
Comment (JunTaoLuo) [inline modules/vaos/app/controllers/vaos/v2/appointments_controller.rb:74]: Keeping this logic in controllers since moving it to services broke mobile. I think they have the expectation that location is set to nil on failure instead of the error message.

**Comment by rjohnson2011** [comment]
Can you get a review from @department-of-veterans-affairs/vfs-vaos before backend approval please?

**Comment by JunTaoLuo** [comment]
Of course! I might make one more commit as well, but I will notify when it's ready for backend review.

File: modules/vaos/app/services/vaos/v2/appointments_service.rb
Comment (JunTaoLuo) [reply modules/vaos/app/services/vaos/v2/appointments_service.rb:97]: FYI I tested out calling `prepare_appointment` here instead of the mish-mash of different function calls and all tests pass so I'm leaning towards just calling `prepare_appointment`. However, this would be a behavior change so I'd still like to double check it before I proceed.

Comment (JunTaoLuo) [reply modules/vaos/app/services/vaos/v2/appointments_service.rb:97]: We decided that further consolidation here requires more discussion and @cferris32 will file a spike to confirm the viability of the change.

File: modules/vaos/app/controllers/vaos/v2/appointments_controller.rb
Comment (JunTaoLuo) [reply modules/vaos/app/controllers/vaos/v2/appointments_controller.rb:74]: I checked with the mobile team and that expectation is due to their business needs so we can't break this behavior for the time being.

**Review by cferris32 — APPROVED** [review]
LGTM

**Review by ryan-mcneil — APPROVED** [review]
👍

**Pull request merged**
